src/Microsoft.AspNet.Routing/Resources.resx

```diff
@@ -189,4 +189,10 @@
   <data name="TemplateRoute_UnescapedBrace" xml:space="preserve">
     <value>In a route parameter, '{' and '}' must be escaped with '{{' and '}}'</value>
   </data>
+  <data name="TemplateRoute_OptionalParameterCanbBePrecededByPeriod" xml:space="preserve">
+    <value>In the complex segment '{0}',  the optional parameter '{1}' is preceded by an invalid segment '{2}'. Only valid literal to precede an optional parameter is a period (.).</value>
```

Comment:
A few suggestions here:
1) Get rid of 'complex' I don't think it's especially meaningful to a user
2) Remove extra space between `{0},` and `the`
3) Change the last sentence here to `Only a period (.) can precede an optional parameter.`